src/Tools/AI Test Toolkit/src/TestSuite/AITTestSuiteMgt.Codeunit.al

```diff
@@ -405,6 +405,27 @@ codeunit 149034 "AIT Test Suite Mgt."
         exit(FeatureNameLbl);
     end;
 
+    internal procedure GetCallerModuleInfo(var CallerModuleInfo: ModuleInfo)
+    var
+        CurrentModuleInfo: ModuleInfo;
+        CallerCallStackModuleInfos: List of [ModuleInfo];
+        ModuleInfo: ModuleInfo;
+    begin
+        // Determine the caller module info from the call stack
+        NavApp.GetCurrentModuleInfo(CurrentModuleInfo);
+        CallerCallStackModuleInfos := NavApp.GetCallerCallstackModuleInfos();
+
+        if CallerCallStackModuleInfos.Count = 1 then
+            CallerModuleInfo := CallerCallStackModuleInfos.Get(1)
+        else
+            // From the list of call stack module infos, get the app id of the last module which is calling the current module
+            foreach ModuleInfo in CallerCallStackModuleInfos do
```

Comment:
Cant we just get the last element of the list based on the length of the list?
I assume it should be on top right?

Reply:
The top one on the is AI TestToolkit iteself. As the caller was coming from AI Test Context, which calls the AI Test Context Impl, and that calls the AITTestSuiteMgt.

If I use "AIT AL Test Suite Mgt", then getting the top should work.